verifier/rest/rest.go

```diff
@@ -142,22 +144,49 @@ func convertChallengeFromREST(chal *confidentialcomputingpb.Challenge) (*verifie
 	}, nil
 }
 
-func convertRequestToREST(request verifier.VerifyAttestationRequest) *confidentialcomputingpb.VerifyAttestationRequest {
-	idTokens := make([]string, len(request.GcpCredentials))
-	for i, token := range request.GcpCredentials {
-		idTokens[i] = string(token)
+func convertTokenOptionsToREST(request *models.TokenOptions) *ccpb.TokenOptions {
+	if request == nil {
+		return nil
+	}
+
+	tokenOptionspb := &ccpb.TokenOptions{
+		Audience: request.Audience,
+		Nonce:    request.Nonces,
 	}
 
-	var tokenType confidentialcomputingpb.TokenType
-	switch request.TokenOptions.TokenType {
+	switch request.TokenType {
 	case "OIDC":
-		tokenType = confidentialcomputingpb.TokenType_TOKEN_TYPE_OIDC
+		tokenOptionspb.TokenType = ccpb.TokenType_TOKEN_TYPE_OIDC
 	case "PKI":
-		tokenType = confidentialcomputingpb.TokenType_TOKEN_TYPE_PKI
+		tokenOptionspb.TokenType = ccpb.TokenType_TOKEN_TYPE_PKI
 	case "LIMITED_AWS":
-		tokenType = confidentialcomputingpb.TokenType_TOKEN_TYPE_LIMITED_AWS
+		tokenOptionspb.TokenType = ccpb.TokenType_TOKEN_TYPE_LIMITED_AWS
+	case "AWS_PRINCIPALTAGS":
+		tokenOptionspb.TokenType = ccpb.TokenType_TOKEN_TYPE_AWS_PRINCIPALTAGS
 	default:
-		tokenType = confidentialcomputingpb.TokenType_TOKEN_TYPE_UNSPECIFIED
+		tokenOptionspb.TokenType = ccpb.TokenType_TOKEN_TYPE_UNSPECIFIED
+	}
+
+	if request.HasAWSPrincipalTagOptions() {
+		tokenOptionspb.TokenTypeOptions = &ccpb.TokenOptions_AwsPrincipalTagsOptions_{
+			AwsPrincipalTagsOptions: &ccpb.TokenOptions_AwsPrincipalTagsOptions{}}
+		if request.PrincipalTagOptions.HasAllowedPrincipalTags() {
+			tokenOptionspb.GetAwsPrincipalTagsOptions().AllowedPrincipalTags = &ccpb.TokenOptions_AwsPrincipalTagsOptions_AllowedPrincipalTags{}
+			if request.PrincipalTagOptions.AllowedPrincipalTags.HasContainerImageSignatures() {
+				tokenOptionspb.GetAwsPrincipalTagsOptions().GetAllowedPrincipalTags().ContainerImageSignatures = &ccpb.TokenOptions_AwsPrincipalTagsOptions_AllowedPrincipalTags_ContainerImageSignatures{
+					KeyIds: request.PrincipalTagOptions.AllowedPrincipalTags.ContainerImageSignatures.KeyIDs,
+				}
+			}
+		}
+	}
```

Comment:
Can you move this to the switch statement above, seem like this only affects AWS_PRINCIPALTAGS  token

Reply:
Also this looks very complicated because of the long variable names, if you can simplify the logic it will be great